net_connection/src/net_connection_thread.rs

```diff
@@ -41,59 +42,84 @@ impl NetConnectionThread {
         Ok(())
     }
 
-    /// create a new NetConnectionThread instance with given handler / worker
+    /// Create a new NetConnectionThread instance with given handler / worker
     pub fn new(
         handler: NetHandler,
         worker_factory: NetWorkerFactory,
         done: NetShutdown,
     ) -> NetResult<Self> {
-        let keep_running = Arc::new(AtomicBool::new(true));
-        let keep_running2 = keep_running.clone();
-
-        let (sender, receiver) = mpsc::channel();
-        Ok(NetConnectionThread {
-            can_keep_running: keep_running,
-            send_channel: sender,
-            thread: thread::spawn(move || {
-                let mut us = 100_u64;
-                let mut worker = worker_factory(handler).unwrap_or_else(|e| panic!("{:?}", e));
-
-                while keep_running2.load(Ordering::Relaxed) {
-                    let mut did_something = false;
-                    receiver
-                        .try_recv()
-                        .and_then(|data| {
+        // Create atomic TRUE
+        let can_keep_running = Arc::new(AtomicBool::new(true));
+        let can_keep_running_shared = can_keep_running.clone();
+        // Create channel
+        let (send_channel, recv_channel) = mpsc::channel();
+
+        let (send_endpoint, recv_endpoint) = mpsc::channel();
+
+        // spawn worker thread
+        let thread = thread::spawn(move || {
+            // Create worker
+            let mut worker = worker_factory(handler).unwrap_or_else(|e| panic!("{:?}", e));
+
+            let endpoint = worker.endpoint();
+            send_endpoint
+                .send(endpoint)
+                .expect("Sending endpoint address should work.");
```

Comment:
I think we might want to explicitly drop the `send_endpoint` variable here so this end of the channel isn't held open for the duration of the thread.

Reply:
good point